ide/editor.settings.storage/src/org/netbeans/modules/editor/settings/storage/fontscolors/ColoringStorage.java

```diff
@@ -365,7 +372,14 @@ public ColoringsReader(FileObject f, String mimePath) {
             try {
                 if (name.equals(E_ROOT)) {
                     // We don't read anythhing from the root element
-                    
+                    colorrefs.clear();;
+                } else if (name.equals(E_COLORREF)) {
+                    String nameAttr = attributes.getValue(A_NAME);
+                    String colorAttr = attributes.getValue(A_COLOR);
+                    String oldRef = colorrefs.put(nameAttr, colorAttr);
```

Comment:
Should null-check and check validity of nameAttr/coloAttr before putting them in the map

Reply:
Add a check that nameAttr is not, in fact, a valid color string. This could lead to very confusing results otherwise, e.g. redefining "ff0000" by accident. Perhaps require color references to use some special syntax, e.g. a color defined as "editorBackground" must be referenced "$editorBackground" (do colorrefs.put("$" + nameAttr, colorAttr) to achieve this). Perhaps also require the names to follow Java identifier requirements, e.g. start with Character.isJavaIdentifierStart and every subsequent character must be Character.isJavaIdentifierPart.

Reply:
Null check is actually done by the parser as both name and color attributes are requested. If they are missing we got a parser exception first before we could check for null.
As of masking predefined colors, I do not see any problem with that. These color definitions is local to the file where they are defined. The whole idea is coming from my attempt to port https://github.com/morhetz/gruvbox/ to NetBeans. In order to support both light and dark mode, I'd just flip the palette definitions.

As of add a distinctive syntax when referring the color definitions. I was thinking about it as well, though I'm natural on that one. The only reason for opting that out is only that less typing makes me happy.